src/aspire/utils/matlab_compat.py

```diff
@@ -94,3 +96,26 @@ def __enter__(self):
     def __exit__(self, *args):
         if self.seed is not None:
             np.random.set_state(random_states.pop())
+
+
+def m_eigsh(*args, **kwargs):
```

Comment:
I don't understand why this function is necessary. Do we have an explicit dependence on the signs of eigenvectors somewhere? If so, where does the convention of flipping the sign to make sure the component with the largest absolute value is positive come from? Is this documented behavior in Matlab somewhere?

Reply:
The eigenvectors of S can change the signs for different repeat runs in the line as shown below: 
 
https://github.com/ComputationalCryoEM/ASPIRE-Python/blob/orientation/src/aspire/orientation/commonline_sync.py#L62

This is a similar problem as `eigh_tridiagonal` found when we included the PSWF 2D basis as noted in line 48 as below:

 https://github.com/ComputationalCryoEM/ASPIRE-Python/blob/develop/src/aspire/basis/pswf_utils.py#L48

Reply:
I am aware of that. That wasn't my question. My first question is how do we depend explicitly on the signs? Normally, when dealing with eigenvectors, the sign doesn't have an effect on the final result due to the way they're used. My second question is where does the convention of making sure the component with largest absolute value is positive come from? Where does Matlab claim to do this?

Reply:
We went through these problem during the group meeting. The different eigenvectors might generate a different set of rotations and we will take a look later.  We agree to rename the `m_eigsh` to `stable_eigsh` and use this function to stabilize the solution.

Reply:
Ok. I checked out an early version and did some comparisons. The first rotation matrix in the saved npy file is listed as below: 
> [ 9.45103184e-01  2.87087608e-01 -1.56079071e-01]
> [-5.63930012e-02  6.13766953e-01  7.87470607e-01]
> [ 3.21869229e-01 -7.35439211e-01  5.96262834e-01]  

When the second run has the eigenvectors with different signs, it becomes as below: 
>  [ 9.45103184e-01  2.87087608e-01 -1.56079071e-01]
>  [ 5.63930012e-02 -6.13766953e-01 -7.87470607e-01]
>  [-3.21869229e-01  7.35439211e-01 -5.96262834e-01]

 But as I guessed during the meeting, The second one can be aligned to the first one by finding the global alignment matrix as below: 
 
>  [ 1.00000000e+00  9.54425519e-16  2.50147996e-15]
>  [ 9.54425519e-16 -1.00000000e+00  4.67052110e-16]
>  [ 2.50147996e-15 -3.21803771e-16 -1.00000000e+00]

So we are OK to the rotation estimation. I think we also prefer to add the wrapper function to get the stable eigenvectors for repeated runs.

Reply:
All right. That makes sense. Thank you for checking!